libyul/optimiser/Suite.cpp

```diff
@@ -383,6 +383,25 @@ void OptimiserSuite::validateSequence(std::string_view _stepAbbreviations)
 	assertThrow(nestingLevel == 0, OptimizerException, "Unbalanced brackets");
 }
 
+bool OptimiserSuite::isEmptyOptimizerSequence(std::string const& _sequence)
+{
+	size_t delimiterCount{0};
+	for (char const step: _sequence)
+		switch (step)
+		{
+		case ':':
+			if (++delimiterCount > 1)
+				return false;
+			break;
+		case ' ':
+		case '\n':
+			break;
+		default:
+			return false;
+		}
+	return true;
+}
```

Comment:
I mean, I'm very much fine with leaving this as is - but given that I think I even already saw some discussion in the channel around this on the side, I can't help but wonder: why the special handling of different counts of colons in here and all the fuzz (the sequence will separately be validated and arguably it even makes more sense to treat ``"::"`` as empty but malformed rather than as non-empty...)?

This could just be (inline):

```suggestion
{
	return ranges::any_of(_sequence, [](auto _step) { return _step != ':' || _step != ' ' || _step != '\n'; });
}
```

in one case - and in the second case for standard json even simpler...